language/move-prover/interpreter/src/concrete/player.rs

```diff
@@ -109,6 +109,21 @@ impl<'env> FunctionContext<'env> {
     // execution
     //
 
+    /// Collect addresses stored in the value recursively
+    fn collect_addresses(val: &BaseValue, addresses: &mut Vec<AccountAddress>) {
+        match val {
+            BaseValue::Address(v) | BaseValue::Signer(v) => {
+                addresses.push(*v);
+            }
+            BaseValue::Vector(vec) | BaseValue::Struct(vec) => {
+                for (_, val) in vec.iter().enumerate() {
+                    Self::collect_addresses(val, addresses);
+                }
+            }
+            _ => (),
```

Comment:
Probably a good idea to fully expand other variants of the `BaseValue` enum.

Reply:
Make sense! Refactored